server/jetstream.go

```diff
@@ -416,6 +417,9 @@ func (s *Server) enableJetStream(cfg JetStreamConfig) error {
 	}
 	s.gcbMu.Unlock()
 
+	// TODO: Not currently reloadable.
+	atomic.StoreInt64(&js.queueLimit, s.getOpts().JetStreamRequestQueueLimit)
```

Comment:
getOpts grabs a lock. So this is now doing lock and atomic.. I do not think either are needed.

Let's consider a global var that is set to the constant 10k to start and opts.go parsing will update both opts (meaning it still can be found in opts) but also updates the global, without any locks..

Reply:
Main problem with a global var is if we have multiple instances in one process, and we'll end up with a race condition if the options code updates the global while one of the JS goroutines is reading it.

In this case, the `getOpts()` call only happens once which is when JetStream is enabled. The atomic isn't really needed right now since even if the opts are reloaded, `js.queueLimit` isn't.

Reply:
Not if we set global var to constant like 

```go
var apiPendingLimit = 10_000
```

Then only if we override it we might see it..

Reply:
ok you are right not in hot path. My bad..